### DISCUSSION TITLE
Making library liter and treeshakable

### DISCUSSION THREAD

**Discussion by mayank1513** [xyflow/xyflow#4333] [New Features]
I see that you are using Zustand for state management. However, you are providing it via React context. 

- Zustand can very well be used without wrappers. And avoiding “client-side” wrappers means you can leverage react18 server components.
- At present library is not treeshakable, if you import component from different files, e.g., import `MiniMap from dist/umd/additional-components/MiniMap`, your app will break → it will be using isolated context provider.

For better understanding, please visit - https://treeshakable.vercel.app/

Proposal
Use `zustand` and wrap it with `treeshakable` to make library liter, more efficient, and treeshakable.

**Comment by moklick** [comment]
I don't understand your points here. 

1. we are using context because we need to implement multiple zustand stores / react flow instances on a page
2. that's not how you should import a component. If you do import { MiniMap } from '@xyflow/react', it gets treeshaked correctly and there won't be any issue with a missing context
